File: src/main/java/com/turchinsky/entities/PassengerEntity.java
package com.turchinsky.entities;


import javax.persistence.*;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotNull;
import java.sql.Date;


@Entity
@Table(name = "passenger")
public class PassengerEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Basic
    @Column
    private String name;

    @Basic
    @Column
    private String surname;

    @Basic
    @Column(name = "birth_date")
    @NotNull
    private Date birthDate;

    @Basic
    @Column(name = "username", nullable = false)
    @NotNull
    private String username;

    @Basic
    @Column(name = "email")
    @NotNull
    @Email
    private String email;

    public PassengerEntity() {
    }

    public PassengerEntity(int id, String name, String surname, Date birthDate, String username, String email) {
        this.id = id;
        this.name = name;
        this.surname = surname;
        this.birthDate = birthDate;
        this.username = username;
        this.email = email;
    }

    //    @SuppressWarnings("JpaAttributeTypeInspection")
//    @OneToMany(mappedBy = "refPassengerEntity")
//    List<TicketEntity> refTicketEntities;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSurname() {
        return surname;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    public Date getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(Date birthDate) {
        this.birthDate = birthDate;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }


//    public List<TicketEntity> getRefTicketEntities() {
//        return refTicketEntities;
//    }
//
//    public void setRefTicketEntities(List<TicketEntity> refTicketEntities) {
//        this.refTicketEntities = refTicketEntities;
//    }

    @Override
    public String toString() {
        return "PassengerEntity{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", surname='" + surname + '\'' +
                ", birthDate=" + birthDate +
                ", username='" + username + '\'' +
                ", email='" + email + '\'' +
                '}';
    }

}

File: src/main/java/com/turchinsky/kt/PassengerEntity.kt
//package com.turchinsky.kt
//
//import javax.persistence.*
//
//@Entity
//@Table(name = "passenger", schema = "public", catalog = "sbb")
//open class PassengerEntity {
//    @get:Id
//    @get:Column(name = "id", nullable = false, insertable = false, updatable = false)
//    var id: Int? = null
//
//    @get:Basic
//    @get:Column(name = "name", nullable = true)
//    var name: String? = null
//
//    @get:Basic
//    @get:Column(name = "surname", nullable = true)
//    var surname: String? = null
//
//    @get:Basic
//    @get:Column(name = "birth_date", nullable = true)
//    var birthDate: java.sql.Date? = null
//
//    @get:Basic
//    @get:Column(name = "username", nullable = false)
//    var username: String? = null
//
//    @get:Basic
//    @get:Column(name = "email", nullable = true)
//    var email: String? = null
//
//    @get:OneToMany(mappedBy = "refPassengerEntity")
//    var refPassengerHasTicketEntities: List<PassengerHasTicketEntity>? = null
//
//    override fun toString(): String =
//            "Entity of type: ${javaClass.name} ( " +
//                    "id = $id " +
//                    "name = $name " +
//                    "surname = $surname " +
//                    "birthDate = $birthDate " +
//                    "username = $username " +
//                    "email = $email " +
//                    ")"
//
//    // constant value returned to avoid entity inequality to itself before and after it's update/merge
//    override fun hashCode(): Int = 42
//
//    override fun equals(other: Any?): Boolean {
//        if (this === other) return true
//        if (javaClass != other?.javaClass) return false
//        other as PassengerEntity
//
//        if (id != other.id) return false
//        if (name != other.name) return false
//        if (surname != other.surname) return false
//        if (birthDate != other.birthDate) return false
//        if (username != other.username) return false
//        if (email != other.email) return false
//
//        return true
//    }
//
//}
//
